river/preprocessing/ordinal.py

```diff
@@ -0,0 +1,91 @@
+import collections
+import math
+import numbers
+
+import numpy as np
+import pandas as pd
+
+from river import base
+
+
+class OrdinalEncoder(base.MiniBatchTransformer):
+    # INFO: makes sense to add None initially? so it'll be easier to spot
+    def __init__(self, handle_unknown="use_reserved_value", unknown_value=1, encoded_missing_value=0):
```

Comment:
How many possible values are there for `handle_unknown`? If it's just a boolean, I rather have an `use_reserved_value_for_unknowns: bool` parameter.

Reply:
Also, wouldn't `-1` make more sense for the `unknown_value`? Isn't it surprising for the first value encountered, as well as unknown values to both be assigned `1`?

Reply:
Which is also making me think, because this method is unsupervised, why don't you do the update logic in `transform_one`? That way there won't be any unknown values, because you update before transforming. I believe we do that in some other places in River

Reply:
> Also, wouldn't `-1` make more sense for the `unknown_value`?

Please see the notes.
I've rejected this idea because there's no '-1' index in an array/tensor
This might seem not very conventional since we're used to '-1' sentinel value but positive-only indexing makes the component more reusable as i've mentioned.

> Isn't it surprising for the first value encountered, as well as unknown values to both be assigned `1`

The idea was to keep 2 options available:
1. same codes for unknown value & missed value
2. different codes for unknown value & missed value

But both codes should provisioned at the start since both should be expected. 
I have no principled idea wether it should be
- unknown = 1 
- missed = 0 
or vice-versa.

Reply:
Ok it's fine, I'm convinced. I'm not sure the +2 works if you specify different values for missing and unknown parameters.

Also, just another question: in your examples you explicitly set missing unknown values to None. But couldn't you instead preserve the sparse nature of the dict and implicitly assume missing keys are None? Not sure I'm being clear